### PR TITLE
common/options: extract mgr and mon options out

### PR DESCRIPTION
**Pull request by tchaikov**
this change is a part of a sequence of changes like
9b5e54d996a771f63b97620d4640c08f6406d843 and
dff996d9e24c6259fffce7e704e47e066c367b6a. because we need to audit
the consumers of an option before extracting it to the dedicated
.yaml.in file, it takes considerable time to get this done.

so let's do this piecemeal.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/options/mon.yaml.in [hunk 1]
```diff
@@ -27,6 +27,11 @@ options:
   default: 4_K
   services:
   - mon
+- name: mon_max_pool_pg_num
+  type: uint
+  level: advanced
+  default: 64_K
+  fmt_desc: The maximum number of placement groups per pool.
```
Comment (neha-ojha) [inline src/common/options/mon.yaml.in:34]: services?

**Review by neha-ojha — APPROVED** [review]

File: src/common/options/mon.yaml.in (hunk 1 above)
Comment (tchaikov) [reply src/common/options/mon.yaml.in:34]: @neha-ojha when an option is ingested, the service property is automatically updated. see https://github.com/ceph/ceph/blob/7fc07dd728b4ee9f83f0e293a5a5ee32020fc7cd/src/common/options/build_options.cc#L40

so unless a given option is shared across multiple services or applications, i think probably we don't need to specify the service explicitly. in this very case, `mon_max_pool_pg_num` does not have `service` specified, so i didn't add it when extracting it from `global.yaml.in` into `mon.yaml.in`.

does this make sense?

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-06-05_02:55:59-rados-wip-kefu-testing-2021-06-05-0940-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-06-08_12:29:52-rados-wip-kefu-testing-2021-06-08-1034-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/50151
- https://tracker.ceph.com/issues/51074

File: src/common/options/mon.yaml.in (hunk 1 above)
Comment (neha-ojha) [reply src/common/options/mon.yaml.in:34]: Yes, thanks for the explanation. Those were missing services, while others were not, so it caught my eye.

**Pull request merged**